sweep/sweeper.go

```diff
@@ -222,6 +222,30 @@ func (p *SweeperInput) terminated() bool {
 	}
 }
 
+// isMature returns a boolean indicating whether the input has a timelock that
+// has been reached or not. The locktime found is also returned.
+func (p *SweeperInput) isMature(currentHeight uint32) (bool, uint32) {
```

Comment:
maybe an option would be better as the return value, so we don't need the boolean, just fn.Option[uint32] ?

Reply:
that would be a big no from my end - I think we are using `fn.Option` in the wrong way, will post more in the dev chat.